systems/common/pushgatewayMetrics/pushgatewayMetrics.go

```diff
@@ -0,0 +1,192 @@
+			prometheus.GaugeOpts{
+				Name:        config.Name,
+				Help:        config.Details,
+				ConstLabels: m.Labels,
+			},
+			customLables,
+		)
+
+	case MetricCounter:
+		m.collector = prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Name:        config.Name,
+				Help:        config.Details,
+				ConstLabels: m.Labels,
+			},
+			customLables,
+		)
+
+	case MetricSummary:
+		m.collector = prometheus.NewSummaryVec(
+			prometheus.SummaryOpts{
+				Name:        config.Name,
+				Help:        config.Details,
+				ConstLabels: m.Labels,
+			},
+			customLables,
+		)
+
+	case MetricHistogram:
+		m.collector = prometheus.NewHistogramVec(
+			prometheus.HistogramOpts{
+				Name:        config.Name,
+				Help:        config.Details,
+				ConstLabels: m.Labels,
+				Buckets:     config.Buckets,
+			},
+			customLables,
+		)
+	default:
+		log.Errorf("Metric %s type %s not supported", config.Name, config.Type)
+		return fmt.Errorf("metric %s type %s not supported", config.Name, config.Type)
+	}
+
+	return nil
+}
+
+func (m *Metrics) SetMetric(value float64, labels prometheus.Labels) error {
+	switch met := m.collector.(type) {
+	case *prometheus.GaugeVec:
+		met.With(labels).Set(value)
+	case *prometheus.CounterVec:
+		met.With(labels).Inc()
+	case *prometheus.SummaryVec:
+		met.With(labels).Observe(value)
+	case *prometheus.HistogramVec:
+		met.With(labels).Observe(value)
+	default:
+		return fmt.Errorf("metric type %s not supported", m.Type)
+	}
+	return nil
+}
+func PushMetrics(pusMetricHost string, metrics []MetricConfig, metriJobName string) {
+
+	labelDimensions := make([]string, 0, len(metrics[0].Labels))
+	for key := range metrics[0].Labels {
+		labelDimensions = append(labelDimensions, key)
+	}
+
+	for _, metric := range metrics {
+		if _, ok := metricCollectors[metric.Name]; !ok {
+			// Metric does not exist, create a new one
+			newMetric := NewMetrics(metric.Name, metric.Type)
+			if err := newMetric.InitializeMetric(metric.Name, MetricConfig{
+				Name:   metric.Name,
+				Type:   metric.Type,
+				Labels: metric.Labels,
+			}, labelDimensions); err != nil {
+				log.Errorf("Failed to initialize metric %s: %v", metric.Name, err)
+				continue
+			}
+			metricCollectors[metric.Name] = []*Metrics{newMetric}
+		}
+
+		for _, m := range metricCollectors[metric.Name] {
+			if err := m.SetMetric(metric.Value, metric.Labels); err != nil {
+				log.Errorf("Failed to set metric %s value: %v", metric.Name, err)
+				continue
+			}
+		}
+	}
+
+	pusher := push.New(pusMetricHost, metriJobName)
+	for _, metrics := range metricCollectors {
+		for _, m := range metrics {
+			pusher.Collector(m.collector)
+		}
+	}
+	if err := pusher.Push(); err != nil {
+		log.Errorf("Could not push metrics to Pushgateway: %s", err.Error())
+	}
+}
+
+func CollectAndPushSimMetrics(pusMetricHost string, configMetrics []MetricConfig, selectedMetric string, Value float64, Labels map[string]string, systemName string) error {
+	var selectedMetrics []MetricConfig
```

Comment:
Does this needs to be an array fi we are only getting one metric?

Reply:
Yes, that's correct. Since we've set up the "before send" function in `config`, it's necessary as the config as already all the labels. If we change it from an array, we'll have to manually set each metric for every request.

Reply:
var selectedMetrics []MetricConfig  this is anyway a local variable and will be set every time you call CollectAndPushSimMetrics function. Since you are calling this with one metric name argument " selectedMetric string" which means at a time only one metric can be pushed. So I don't see a need of array. Let me know If I am missing anuything.

Reply:
You right.

Reply:
You're right .It need to be a single metric .